src/types/list.rs

```diff
@@ -589,6 +613,31 @@ impl<'py> BoundListIterator<'py> {
         }
     }
 
+    /// # Safety
+    ///
+    /// On the free-threaded build, caller must verify they have exclusive
+    /// access to the list by holding a lock or by holding the innermost
+    /// critical section on the list.
+    #[inline]
+    #[cfg(not(Py_LIMITED_API))]
```

Comment:
you should also add branches to `nth` and `nth_back` that use `get_item` instead of `get_item_unchecked` on the limited API